### ISSUE TITLE
Download failed

### ISSUE THREAD

**Issue by logion2000** [elementdavv/internet_archive_downloader#20]
I used Firefox. I don't why the download failed. I have tried several times. Also, when I used MS Edge, the Quality button does not show up. See the second screenshot below.

![image](https://github.com/elementdavv/internet_archive_downloader/assets/89305695/da0fa6fe-5c6a-4eda-b795-9701825a88bb)

![image](https://github.com/elementdavv/internet_archive_downloader/assets/89305695/0f55fcad-685c-437c-bc5e-6a334c1f6bb5)

**Comment by elementdavv** [comment]
About "cross origin request blocked" error on Firefox, I have tested the extension on every platforms I can find such as Windows/Mac/Ubuntu/Fedora/Arch, and it works out of the box for every fresh platform installations. But there are persons who report the error as you. May be other installed applications or system security policies have impacted the Firefox functions, I have no way to debug it without the environments that lead to the error. 
For your situation, you can make some tries, such as disable other extensions, disable firewall and other security settings, or create and use a new Firefox profile. If every effort fails, then I suggest you to change to Chrome family browsers.

As for the no Quality button error in Edge, it is so ridiculous. The two buttons should be both showing or not showing. Can you take a look at the developer console page, may be some errors happened.

**Comment by logion2000** [comment]
Here is the screenshot of the developer console. Hopefully you can find what went wrong.
![image](https://github.com/elementdavv/internet_archive_downloader/assets/89305695/e9a487db-53a0-4fed-a82c-3d076594b476)

![image](https://github.com/elementdavv/internet_archive_downloader/assets/89305695/81a62093-73a2-43c9-9a62-8315a7e5f78c)

**Comment by elementdavv** [comment]
> Here is the screenshot of the developer console. Hopefully you can find what went wrong.
The messages on your screenshot do not related to the extension.
Do you install the extension from Edge addons repository? It is not up to date and has only one Download button. You should install the latest version from Chrome webstore.

**Comment by asheroto** [comment]
Having the same issue for some reason.  Using the latest version of Chrome.

![image](https://github.com/elementdavv/internet_archive_downloader/assets/49938263/11380ca0-6a2c-47de-94b6-ee5d93343409)

https://archive.org/details/iamdiscourses00king/page/2/mode/2up

**Comment by elementdavv** [comment]
> Having the same issue for some reason. Using the latest version of Chrome.
> Just installed the Chrome extension from the web store.
> Version 0.6.3
> Click 'Download' a few times and got this...
> Would the size matter? Trying to download a nearly 400 page PDF on a 1 Gbps connection as a test.

You are using Chrome with the extension v0.6.3 ?
So how do you operate to get a request to elementdavv.github.io ? This domain is only used by the extension for Firefox. The extension for Chrome will work without elementdavv.github.io.

For the File download blocked issue, please provide a complete messages from the console page.

The file size does not matter. It can download a PDF file of several thousand pages and several GB size.

**Comment by asheroto** [comment]
I was mistaken when I said I used Chrome. I am using **Brave**, which is Chromium-based. I found the problem and it's not the extension's fault! With Brave, it has built-in ad-blocking and other security items that help protect users on the web. When Brave Shields are on, the extension fails, but if you turn them off like so, the extension works fine. Sorry about that! 😳

![image](https://github.com/elementdavv/internet_archive_downloader/assets/49938263/8e0f9c43-a3c2-4db0-872a-629c4eb21708)

So heads up, **if you have Brave users they need to turn off the Brave shields.**


<details><summary>Working with Brave Shields Disabled</summary>
<p>

![image](https://github.com/elementdavv/internet_archive_downloader/assets/49938263/329c799b-02da-45ce-b74b-648c220a1ccf)

</p>
</details> 

<details><summary>Not Working with Brave Shields Enabled</summary>
<p>

![image](https://github.com/elementdavv/internet_archive_downloader/assets/49938263/1eeaba60-afea-408d-ae50-58dd037b029c)

</p>
</details> 

---

If it's not working with Edge or other browsers like the OP mentioned, perhaps it is due to some security, blocking fingerprinting or third-party cookies similar to what Brave shields does...

![image](https://github.com/elementdavv/internet_archive_downloader/assets/49938263/e7e65448-7569-4f54-86df-e681b352ddfe)

Hope that helps! As for me, I'm set, but other users may have a separate issue if they've disabled some of these extra security items in their browser.

**Comment by elementdavv** [comment]
>Hope that helps! As for me, I'm set, but other users may have a separate issue if they've disabled some of these extra security items in their browser.

Yes. You are right. And you can simply enable item 'File System Access API' in 'brave://flags' page without touching Shields.

**Comment by asheroto** [comment]
Gotcha. But that would enable it for all sites, right? The shields only affect the individual site when you toggle it...

**Comment by elementdavv** [comment]
> Edge, download PDF file but zero byte content... and i try the various descripted methods.

What is the pdf url? Did you try Chrome?

**Comment by eelalzep** [comment]
I'm using Firefox Android and downloading keeps failing, and when i decide to go in private mode it pops up this![Screenshot_2024-03-22-08-02-47-520_org.mozilla.firefox.jpg](https://github.com/elementdavv/internet_archive_downloader/assets/77693896/ba222f98-2219-4cc5-b62a-2a544ceec181)
